src/S3.ts

```diff
@@ -170,7 +170,25 @@ class S3Bucket {
             return false
         }
     }
-
+    async containsFileWithPrefix(prefix: string) {
+        let s3 = (await this.s3Client())
+        try {
+            
+            let response = await s3.listObjectsV2({
+                Bucket: this.name,
+                Prefix : prefix,
+                MaxKeys: 1
+            }).promise();
+            if(response.Contents && response.Contents.length>0) {
+                return true;
+            } else {
+                return false;
+            }
+        } catch (e) {
+            expect(e.code).to.equal('AccessDenied')
```

Comment:
This should not catch any errors IMO.

If you don't have access - this method should throw an error - otherwise could lead to false positives